elephant/spike_train_processing.py

```diff
@@ -0,0 +1,317 @@
+    ----------
+    spiketrains : list of neo.SpikeTrains
+        a list of neo.SpikeTrains objects. These spike trains should have been
+        recorded simultaneously.
+    sampling_rate : pq.Quantity
+        Sampling rate of the spike trains.
+    spread : int, optional
+        Number of bins in which to check for synchronous spikes.
+        Spikes that occur separated by `spread - 1` or less empty bins are
+        considered synchronous.
+          * `spread = 0` corresponds to a bincount accross spike trains.
+          * `spread = 1` corresponds to counting consecutive spikes.
+          * `spread = 2` corresponds to counting consecutive spikes and
+            spikes separated by exactly 1 empty bin.
+          * `spread = n` corresponds to counting spikes separated by exactly
+            or less than `n - 1` empty bins.
+        Default: 0
+
+    Returns
+    -------
+    complexity_intervals : neo.Epoch
+        An epoch object containing complexity values, left edges and durations
+        of all intervals with at least one spike.
+
+    Raises
+    ------
+    ValueError
+        When `t_stop` is smaller than `t_start`.
+
+    Examples
+    --------
+    Here the behavior of
+    `elephant.spike_train_processing.precise_complexity_intervals` is shown, by
+    applying the function to some sample spiketrains.
+
+    >>> import neo
+    >>> import quantities as pq
+
+    >>> sampling_rate = 1/pq.ms
+    >>> st1 = neo.SpikeTrain([1, 4, 6] * pq.ms, t_stop=10.0 * pq.ms)
+    >>> st2 = neo.SpikeTrain([1, 5, 8] * pq.ms, t_stop=10.0 * pq.ms)
+
+    >>> # spread = 0, a simple bincount
+    >>> ep1 = precise_complexity_intervals([st1, st2], sampling_rate)
+    >>> print(ep1.array_annotations['complexity'].flatten())
+    [2 1 1 1 1]
+    >>> print(ep1.times)
+    [0.  3.5 4.5 5.5 7.5] ms
+    >>> print(ep1.durations)
+    [1.5 1.  1.  1.  1. ] ms
+
+    >>> # spread = 1, consecutive spikes
+    >>> ep2 = precise_complexity_intervals([st1, st2], sampling_rate, spread=1)
+    >>> print(ep2.array_annotations['complexity'].flatten())
+    [2 3 1]
+    >>> print(ep2.times)
+    [0.  3.5 7.5] ms
+    >>> print(ep2.durations)
+    [1.5 3.  1. ] ms
+
+    >>> # spread = 2, consecutive spikes and separated by 1 empty bin
+    >>> ep3 = precise_complexity_intervals([st1, st2], sampling_rate, spread=2)
+    >>> print(ep3.array_annotations['complexity'].flatten())
+    [2 4]
+    >>> print(ep3.times)
+    [0.  3.5] ms
+    >>> print(ep3.durations)
+    [1.5 5. ] ms
+
+    """
+    _check_consistency_of_spiketrainlist(spiketrains)
+
+    bin_size = 1 / sampling_rate
+
+    bst = conv.BinnedSpikeTrain(spiketrains,
+                                binsize=bin_size)
+
+    bincount = np.array(bst.to_sparse_array().sum(axis=0)).squeeze()
+
+    if spread == 0:
+        bin_indices = np.nonzero(bincount)[0]
+        complexities = bincount[bin_indices]
+        left_edges = bst.bin_edges[bin_indices]
+        right_edges = bst.bin_edges[bin_indices + 1]
```

Comment:
are you sure that the right edges always exist? I mean, cannot `bin_edges[bin_indices + 1]` be outside of the range?